packages/venia-ui/lib/components/AddressBookPage/addEditDialog.js

```diff
@@ -0,0 +1,178 @@
+import React from 'react';
+import { bool, func, object, shape } from 'prop-types';
+import { useIntl } from 'react-intl';
+
+import { mergeClasses } from '@magento/venia-ui/lib/classify';
+import { isRequired } from '@magento/venia-ui/lib/util/formValidators';
+
+import Checkbox from '@magento/venia-ui/lib/components/Checkbox';
+import Country from '@magento/venia-ui/lib/components/Country';
+import Dialog from '@magento/venia-ui/lib/components/Dialog';
+import Field from '@magento/venia-ui/lib/components/Field';
+import FormError from '@magento/venia-ui/lib/components/FormError';
+import Postcode from '@magento/venia-ui/lib/components/Postcode';
+import Region from '@magento/venia-ui/lib/components/Region';
+import TextInput from '@magento/venia-ui/lib/components/TextInput';
+import defaultClasses from './addEditDialog.css';
+
+const AddEditDialog = props => {
+    const {
+        formErrors,
+        formProps,
+        handleCancel,
+        handleConfirm,
+        isBusy,
+        isEditMode,
+        isOpen,
+        numAddresses
+    } = props;
+
+    const { formatMessage } = useIntl();
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+    // Disable the checkbox if this is the first address being added
+    // or the only address being edited.
+    // The box is auto-checked in these cases by useAddressBookPage.js.
+    const isCheckboxDisabled =
+        (!isEditMode && numAddresses === 0) ||
+        (isEditMode && numAddresses === 1);
```

Comment:
So this is worth calling out: If this is the first address being created, the talon will check the "make this my default address" checkbox and this code will disable it.

The same holds true when editing the only address.

The backend does allow a single address that is _not_ the default, which is a little funky.
In fact, you can't delete an address that is marked as the default, so this may complicate the implementation of removing an address (you'd have to set it as not the default first).

I'm open to suggestions on which way is better:
1. This, where the only address is assumed to be the default but removing it would require two calls (one to mark it as not the default, the second to actually delete it)
2. Don't automatically mark addresses as default. Always let the user control this.

Reply:
I think the former is probably more in line with UX expectations. It's not a big deal if the delete callback takes slightly more time to complete.

Reply:
We talked about this in team time today. Resolved in https://github.com/magento/pwa-studio/pull/2879/commits/03880a99fbf40aed7e8cff6eaa7bc2c2a136fd06.